src/SelfRoleManager.ts

```diff
@@ -272,6 +281,25 @@ export class SelfRoleManager extends EventEmitter {
         break;
     }
   }
+
+  #listenRoleChanges(rolesToEmojis: RoleToEmojiData[]) {
+    this.client.on(Events.GuildMemberUpdate, async (oldMember, newMember) => {
+      const rolesToRemove = rolesToEmojis.filter(
+        (rte) =>
+          rte.dependencies?.some((dep) => {
+            const role = dep instanceof Role ? dep.id : dep;
+            return oldMember.roles.cache.has(role) && !newMember.roles.cache.has(role);
```

Comment:
Avoid relying on the cache as much as possible. I'd rather `.fetch()` the information before (as the fetch method checks in the cache before making an API call if necessary) to avoid issues with missing data!

Reply:
Thinking about it again, I'm pretty sure the `GuildMemberRoleManager` does not have a `fetch` method. Which means we should have a `GuildMember` object with a proper cache. So this should be OK!